source/agora/node/Registry.d

```diff
@@ -216,10 +241,17 @@ public class NameRegistry: NameRegistryAPI
         TYPE payload_type = this.ensureValidPayload(data,
             this.zones[ZoneIndex.Validator].get(data.public_key));
 
-        if (this.zones[ZoneIndex.Validator].type == ZoneType.secondary)
+        if (sig !is Signature.init)
```

Comment:
Since `getValidatorInternal` should not be expected for the secondary, this case cannot happen.